### PR TITLE
Appwrite Lite

### PR DESCRIPTION
**Pull request by lohanidamodar**
Hey,

1. Have a basic dockerfile that uses appwrite/appwrite:dev as base image
2. Have installed supervisord and redis
3. Disabled stats, influxdb and telegraf
3. A simple supervisord.conf to start Appwrite services, with logs in proper files
4. A simple redis.conf, taken from redis-alpine image
5. docker-compose.yml to load services
6. .travis.yml for travis check taken from main appwrite repo

### CODE REVIEW

File: supervisord.conf [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+[program:worker-usage]
+command=worker-usage
+autostart=true
+autorestart=true
+priority=5
+stdout_events_enabled=true
+stderr_events_enabled=true
+stopsignal=QUIT
+startretries=10
+stdout_logfile=/var/log/appwrite.log
```
Comment (eldadfux) [inline supervisord.conf]: It will be nice and easier to debug specific issues if we'll use a separate log file for each process stdout & stderr logs.

Comment (lohanidamodar) [reply supervisord.conf]: I have made the changes.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM appwrite/appwrite:dev
+
+LABEL maintainer="team@appwrite.io"
```
Comment (eldadfux) [inline Dockerfile:3]: Would be a nice idea to add the `_APP_EDITION` env var and overwrite the default `community` value with `community-lite` we can use it later to distinguish usage stats.

Comment (eldadfux) [reply Dockerfile:3]: In addition, maybe we can add some of the .env file settings directly in this Dockerfile

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+version: '3'
+
+services:  
+  appwritelite:
+    container_name: appwritelite
```
Comment (eldadfux) [inline docker-compose.yml]: appwrite-lite would be more consistent to our naming conventions in this context

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    environment:
+      - MYSQL_ROOT_PASSWORD=password
+      - MYSQL_DATABASE=${_APP_DB_SCHEMA}
+      - MYSQL_USER=${_APP_DB_USER}
+      - MYSQL_PASSWORD=${_APP_DB_PASS}
+    command: 'mysqld --innodb-flush-method=fsync'
+  
+
+networks:
+  appwritelite:
```
Comment (eldadfux) [inline docker-compose.yml]: appwrite-lite

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+volumes:
+  appwritelite-mariadb:
```
Comment (eldadfux) [inline docker-compose.yml]: appwrite-lite

File: root/etc/supervisord.conf [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+; (supervisorctl/web interface) to work, additional interfaces may be
+; added by defining them in separate rpcinterface: sections
+[rpcinterface:supervisor]
+supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
+
+[supervisorctl]
+serverurl=unix:///tmp/supervisor.sock ; use a unix:// URL  for a unix socket
+
+[program:appwrite]
+command=php /usr/src/code/app/http.php -dopcache.preload=opcache.preload=/usr/src/code/app/preload.php
```
Comment (eldadfux) [inline root/etc/supervisord.conf:26]: I would add a list of all the logs files and how to view them if needed.
`docker-compose exec appwrite-lite cat /path/to/log`.

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FROM appwrite/appwrite:dev
```
Comment (eldadfux) [inline Dockerfile]: Don't forget to update the tag when 0.7 is out

**Review by eldadfux — COMMENTED** [review]
Please also update the README file with explanation about how this edition is different in terms of both features and architecture (monolith vs micorservices). Example of setup (can be manual), links to Appwrite docs, getting started tutorials and more.

File: Dockerfile (hunk 2 above)
Comment (lohanidamodar) [reply Dockerfile:3]: done!

File: .env [hunk 8]
```diff
@@ -0,0 +1,27 @@
+_APP_ENV=development
```
Comment (eldadfux) [inline .env]: Anyway for to reduce this list to bare-minimum? Stuff like `_APP_ENV` already default to best practice production value. I would keep here only *must have* settings. Same for the `.env` file example in the readme.

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,21 @@
+FROM appwrite/appwrite:0.7.0
+
+LABEL maintainer="team@appwrite.io"
+
+RUN \
+  apk update \
+  && apk add --no-cache supervisor redis
+
```
Comment (eldadfux) [inline Dockerfile:8]: Extra space

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+# copy supervisord conf
+ADD root /
+
+# Environment
+ENV _APP_EDITION=community-lite \
+    _APP_STORAGE_ANTIVIRUS=disabled \
+    _APP_REDIS_HOST=localhost \
+    _APP_REDIS_PORT=6379 \
+    _APP_USAGE_STATS=disabled
+
```
Comment (eldadfux) [inline Dockerfile:18]: Extra spaces

File: README.md [hunk 11]
```diff
@@ -1,2 +1,168 @@
...
+- [Whats different](#whats-different)
+- [Installation](#installation)
+- [Getting Started](#getting-started)
+    
+## What's different
+Appwrite lite is simpler and light weight version. So we have removed few functionalities and made some fundamental changes as follows.
+1. All the services are running inside a single container using supervisord.
+2. Services like ClamAV antivirus and InfluxDB and Telegraf for usage stats have been removed.
+
+In case of features, all the features of Appwrite is present except for usage stats and Antivirus.
```
Comment (eldadfux) [inline README.md:32]: Maybe we should add a section: `Who is it good for?` also explaining why use this over regular version. I would also mention on the the `What's different` version, which services are omitted, also explaining why each one was omitted.

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: .env (hunk 8 above)
Comment (lohanidamodar) [reply .env]: I don't see anything beside _APP_ENV can be removed?
1. can't remove DB related env,
2. Better not to remove SMTP related stuff
3. May be can we remove Storage limit?
4. Functions env?

Comment (eldadfux) [reply .env]: Actually you can remove almost everything. Any setting that can already be set in the dockerfile itself can be removed from here. Leave only stuff that the user is likely to want and change (or must change).

For example, no real need in the Cloud Functions stuff for a quick setup. _APP_SYSTEM_RESPONSE_FORMAT is likely to not change on first setup. Storage and abuse limits as well. Also avoid any Appwrite emails, this gets our inbox busy.

**Pull request merged**